### PR TITLE
Adding Java runtime.

### LINKED ISSUE #93: Add Support For Java Runtimes
Per what I see in https://github.com/serverless/serverless-openwhisk/blob/master/compile/functions/runtimes/index.js it does not appear that this plugin support Java runtimes. 

OpenWhisk does support this runtime: https://github.com/apache/incubator-openwhisk-runtime-java

### PR DESCRIPTION
**Pull request by allen-servedio**
Closes #93 

@jthomas - Okay, here is my attempt at adding the Java runtime. Apologies if there are some obvious NodeJS coding issues. Just let me know what you want altered.

I installed it locally and pushed up a simple Java action and was able to call it.

### CODE REVIEW

**Comment by jthomas** [comment]
@allen-servedio Fast work on getting this implemented! 

I'll code review tomorrow and let you know if I find anything. 

Can you add docs to the README for Java support? Follow the style here: 
https://github.com/serverless/serverless-openwhisk#writing-functions---php

Once this is ready, we should drop a PR to the main framework project. This will include adding docs which get imported on hosted on their site, a template project for the Java runtime and examples in the `serverless/examples` project to show people how to use it. 

Thanks again for doing this! Congrats on the first PR to this project. Since you know you're way around the code-base now, feel free to open issues and send PRs for other stuff you find!

**Comment by allen-servedio** [comment]
Okay @jthomas, pushed the docs (sorry I missed them initially). Let me know what changes you want and generally what you want in the PR for the examples, etc...

File: README.md [hunk 1]
```diff
@@ -376,6 +376,125 @@ Binaries produced by the Swift build process must be generated for the correct p
...
+In the `serverless.yaml` file (see below), the `handler` property is the uberjar produced by calling `mvn clean package`. To specify the name of the class containing the `main` function, add an `annotation` called `main_class` (this should be the fully qualified class name). If you do not provide this `annotation`, it will look for a class in the default package called `Main`.
+
+```yaml
+service: my-java-service
+provider:
+  name: openwhisk
+  runtime: java
+functions:
+  hello:
+    handler: target/hello-world-1.0.jar
```
Comment (jthomas) [inline README.md]: I'd prefer to provide both the JAR file and the main class in the `handler` parameter. This keeps it consistent with the other runtimes which use a `file.method` style. For this runtime, we would adopt the following:
```
    handler: target/hello-world-1.0.HelloWorld
```

This should be a small change to your PR - I think if you back out the convertHandlerToPath & calculateFunctionMain fns it should just work?

**Comment by jthomas** [comment]
This PR is looking really good - great doc updates! 

Everything looks fine apart from a small change to how we define the `main` to be consistent with the other runtimes. If you can get that done, I'll merge this and work on a new release to get this great change out asap. 

Thanks for the work on this 👍

File: README.md (hunk 1 above)
Comment (allen-servedio) [reply README.md]: That is going to get a little gnarly if they follow the standard practice of NOT using the default package and actually place their class into a package. For example, if they put the class in a package called `com.company.app` then the handler will look like this:

```
handler: target/hello-world-1.0.com.company.app.HelloWorld
```

That is visually a bit hard to parse (and I THINK breaks the code you were referring to). I suppose you could do something like this:

```
handler: target/hello-world-1.0.jar:com.company.app.HelloWorld
```

The reason I used annotation in this pull request was I saw that was how you marked an action as a web action... So I was thinking annotations could be used to provide further refinements of how actions needed to be deployed. That being said, I understand what you mean by consistency, so definitely open to changing that part.

Comment (jthomas) [reply README.md]: I can see that might get a bit long. `handler: target/hello-world-1.0.jar:com.company.app.HelloWorld` is definitely a bit better style. 

[Annotations](https://github.com/apache/incubator-openwhisk/blob/master/docs/annotations.md) are actually passed through to the platform when creating the actions. They aren't used for meta-data inside the framework if you see what I mean? 

I agree it does get a bit long but think the second style (with `:`) is the best compromise between consistency and readability.

**Comment by jthomas** [comment]
This looks perfect, thanks for the changes. I'll make a note to get a release out with this in soon.
